tasks/connectors/edgescan/lib/edgescan_location_specifier.rb

```diff
@@ -24,21 +25,81 @@ def location
         end
 
         def to_kenna_asset
-          base = { "external_id" => external_asset_id, "tags" => asset.tags }
+          case type
+          when "url"
+            { **kenna_asset_base, **{ "url" => location }, **host_information(location) }
+          when "hostname"
+            { **kenna_asset_base, **{ "hostname" => location }, **host_information(location) }
+          when "ip"
+            { **kenna_asset_base, **{ "ip_address" => location }, **host_information(location) }
+          when "block"
+            to_kenna_assets_from_block
+          when "cidr"
+            to_kenna_assets_from_cidr
+          end
+        end
+
+        private
+
+        def to_kenna_assets_from_block
+          split_ip = location.split(".")
+          start_of_ip = split_ip[0..2].join(".")
+          lower_bound, upper_bound = split_ip.last.split("-")
+          (lower_bound.to_i..upper_bound.to_i).each_with_object([]) do |host_field, kenna_assets|
+            new_ip = "#{start_of_ip}.#{host_field}"
+            base = { "external_asset_id" => "ES#{asset['id']} #{new_ip}", "ip_address" => new_ip }
+            kenna_assets << { **kenna_asset_base, **base, **host_information(new_ip) }
+          end
+        end
 
-          base["url"] = location if type == "url"
-          base["hostname"] = location if type == "hostname"
-          base["ip_address"] = location if type == "ip"
-          base["ip_address"] = location.split("-").first if type == "block"
-          base["ip_address"] = location.split("/").first if type == "cidr"
+        def to_kenna_assets_from_cidr
```

Comment:
This will create dummy assets that were never picked up by Edgescan, right?

Reply:
This will create a kenna asset for each of the locations in a cidr or block ip address range, whether the location has an associated vulnerability or not.

Reply:
This doesn't make sense to me as we'd be creating assets in Kenna for assets that WERE NOT identified by Edgescan - they're just in a particular IP address range.

Reply:
Removed in ffb6b20.